socorro/unittest/external/boto/__init__.py

```diff
@@ -0,0 +1,28 @@
+from configman import ConfigurationManager
+import mock
+
+
+def get_config(cls, values_source=None):
+    """Generates config based on the required config of the cls
+
+    :arg cls: a configman-enhanced class
+    :arg values_source: dict of non-default configuration to change
+
+    :returns: a configman config object
+
+    """
+    values_source = values_source or {}
+
+    conf = cls.get_required_config()
+    conf.add_option('logger', default=mock.Mock())
+    conf.add_option('metrics', default=mock.Mock())
+
+    cm = ConfigurationManager(
+        [conf],
+        app_name='testapp',
+        app_version='1.0',
+        app_description='TelemetryBotoS3CrashStorage test',
+        values_source_list=[values_source],
+        argv_source=[],
+    )
+    return cm.get_config()
```

Comment:
We had a bunch of code that does something along these lines, but did it worse. In most cases, they built the config object manually which requires you to specify everything instead of building it from the `required_config` of the class which will use the defaults. I'd much rather do the latter since as we change defaults, we can see how it affects our tests. Further, this reduces the amount of configuration code we've got floating around.

It's possible something like this exists in the tests already. If not, we probably want to hoist it upwards and use it everywhere.